bld/CLMBuildNamelist.pm

```diff
@@ -2779,7 +2780,7 @@ sub setup_logic_irrigation_parameters {
      my $var;
      foreach $var ("irrig_min_lai", "irrig_start_time", "irrig_length",
                    "irrig_target_smp", "irrig_depth", "irrig_threshold_fraction",
-                   "limit_irrigation_if_rof_enabled") {
+                   "limit_irrigation_if_rof_enabled","use_groundwater_irrigation") {
         add_default($opts, $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, $var);
      }
 
```

Comment:
You should add a check that use_groundwater_irrigation and lower_boundary_condition==3|4 aren't both set. The FORTRAN code checks this so it should be checked sooner in build-namelist as well. It's good to have it check for a valid namelist as soon as possible.

setup_logic_hydrology_params is already called before setup_logic_irrigation_parameters.

You just add something like this:

``` perl
my $lower = $nl->get_value( 'lower_boundary_condition'  );
if ( ($lower == 3 || $lower == 4)) & (&value_is_true($nl->get_value( 'use_groundwater_irrigation'  )) ) {
 $log->fatal_error("use_groundwater_irrigation can only be used when lower_boundary_condition is NOT 3 or 4");
}
```

Reply:
Okay, I added this and tested by adding use_groundwater_irrigation to user_nl_clm for both a clm50 case (works) and clm45 case (dies with the error).